src/components/DecisioningEngine/createSubscribeRulesetItems.js

```diff
@@ -27,20 +28,21 @@ const validateOptions = ({ options }) => {
 };
 
 export default () => {
-  let subscriptionHandler;
+  const subscription = createSubscription();
   let surfacesFilter;
   let schemasFilter;
 
   const run = ({ surfaces, schemas, callback }) => {
-    subscriptionHandler = callback;
+    const unsubscribe = subscription.add(callback);
     surfacesFilter = surfaces instanceof Array ? surfaces : undefined;
     schemasFilter = schemas instanceof Array ? schemas : undefined;
```

Comment:
You're using the same filters for all subscriptions. I'd expect the filters to be specific to this callback.

Reply:
wow, glad you caught this. 🤦‍♂️